c++/src/kj/table.c++

```diff
@@ -279,13 +279,14 @@ size_t BTreeImpl::verifyNode(size_t size, FunctionParam<bool(uint, uint)>& f,
     auto n = leaf.size();
     for (auto i: kj::zeroTo(n)) {
       KJ_ASSERT(*leaf.rows[i] < size, n, i);
-      if (i + 1 < n) {
-        KJ_ASSERT(f(*leaf.rows[i], *leaf.rows[i + 1]),
-            n, i, leaf.rows[i], leaf.rows[i + 1]);
-      } else if (maxRow != nullptr) {
-        KJ_ASSERT(leaf.rows[n-1] == maxRow);
+      if (i > 0) {
+        KJ_ASSERT(f(*leaf.rows[i - 1], *leaf.rows[i]),
+            n, i, leaf.rows[i - 1], leaf.rows[i]);
       }
     }
+    if (maxRow != nullptr) {
+      KJ_ASSERT(leaf.rows[n-1] == maxRow, n);
```

Comment:
`n` is guaranteed to be > 0 here? (This would be a preexisting problem if it were one, of course.)